mani_skill/envs/tasks/control/hopper.py

```diff
@@ -0,0 +1,266 @@
+            gpu_memory_cfg=GPUMemoryConfig(
+                found_lost_pairs_capacity=2**25, max_rigid_patch_count=2**18
+            ),
+            scene_cfg=SceneConfig(
+                solver_position_iterations=15, solver_velocity_iterations=15
+            ),
+            # sim_freq=50,
+            control_freq=25,
+        )
+
+    @property
+    def _default_sensor_configs(self):
+        return self.agent._sensor_configs
+
+    @property
+    def _default_human_render_camera_configs(self):
+        return self.agent._sensor_configs
+
+    def _load_scene(self, options: dict):
+        loader = self.scene.create_mjcf_loader()
+        articulation_builders, actor_builders, sensor_configs = loader.parse(MJCF_FILE)
+        for a in actor_builders:
+            a.build(a.name)
+
+        # load in the ground
+        self.ground = build_ground(
+            self.scene,
+            floor_width=2,
+            floor_length=100,
+            altitude=-0.075,
+            xy_origin=(50 - 2, 0),
+        )
+
+    def _initialize_episode(self, env_idx: torch.Tensor, options: Dict):
+        with torch.device(self.device):
+            b = len(env_idx)
+            # qpos sampled same as dm_control, but ensure no self intersection explicitly here
+            random_qpos = torch.rand(b, self.agent.robot.dof[0])
+            q_lims = self.agent.robot.get_qlimits()
+            q_ranges = q_lims[..., 1] - q_lims[..., 0]
+            random_qpos *= q_ranges
+            random_qpos += q_lims[..., 0]
+
+            # overwrite planar joint qpos - these are special for planar robots
+            # first two joints are dummy rootx and rootz
+            random_qpos[:, :2] = 0
+            # y is axis of rotation of our planar robot (xz plane), so we randomize around it
+            random_qpos[:, 2] = torch.pi * (2 * torch.rand(b) - 1)  # (-pi,pi)
+            self.agent.robot.set_qpos(random_qpos)
+
+    def evaluate(self):
+        return dict()
+
+    @property  # dm_control mjc function adapted for maniskill
+    def height(self):
+        """Returns relative height of the robot"""
+        return (
+            self.agent.robot.links_map["torso"].pose.p[:, -1]
+            - self.agent.robot.links_map["foot_heel"].pose.p[:, -1]
+        ).view(-1, 1)
+
+    @property  # dm_control mjc function adapted for maniskill
+    def subtreelinvelx(self):
+        """Returns linvel x component of robot"""
+        return self.agent.robot.get_qvel()[:, 0].view(-1, 1)
+
+    # dm_control mjc function adapted for maniskill
+    def touch(self, link_name):
+        """Returns function of sensor force values"""
+        force_vec = self.agent.robot.get_net_contact_forces([link_name])
+        force_mag = torch.linalg.norm(force_vec, dim=-1)
+        return torch.log1p(force_mag)
+
+    # dm_control also includes foot pressures as state obs space
+    def _get_obs_extra(self, info: Dict):
+        obs = dict(
+            toe_touch=self.touch("foot_toe"),
```

Comment:
does dm control with visual data input ever use state information? This data will be part of the observation (along with _get_obs_agent())

Reply:
I am now explicitly overriding _get_obs_state_dict to include entire state (touch + proprioception), so the touch information isn't allowed for rgbd obs mode

Reply:
dm_control uses pixel wrapper, which allows for user to choose either entire state + pixels or only pixels

Reply:
Ok looking at what seems to be standard practice, DrQ and TDMPC both do not use state at all in training, e.g. https://github.com/nicklashansen/tdmpc2/blob/5f6fadec0fec78304b4b53e8171d348b58cac486/tdmpc2/envs/wrappers/pixels.py#L8

Given the tasks are not that hard, we can do image only ppo as well. The way I will do it is to fix the flatten rgbd obsevation wrapper to be like the pixel wrapper, and you just choose if you want rgb, depth, and / or state, and it will handle it accordingly.

Reply:
re this, merge the main branch into your PR. I have updated code to do this exactly. The new ppo rgb script is this:

```
python ppo_rgb.py --exp_name="rgbhopseed1" --env_id="MS-HopperHop-v1" --num_envs=512 --update_epochs=8 --num_minibatches=32 --total_timesteps=20_000_000 --eval_freq=10 --num_eval_steps=600 --num_steps=50 --gamma=0.99 --seed=1 --no_include_state
```
(the hyperparams might not be well tuned).